src/catalog.c

```diff
@@ -651,10 +659,16 @@ ts_catalog_insert_values(Relation rel, TupleDesc tupdesc, Datum *values, bool *n
 }
 
 TSDLLEXPORT void
-ts_catalog_update_tid(Relation rel, ItemPointer tid, HeapTuple tuple)
+ts_catalog_update_tid_only(Relation rel, ItemPointer tid, HeapTuple tuple)
 {
 	CatalogTupleUpdate(rel, tid, tuple);
 	ts_catalog_invalidate_cache(RelationGetRelid(rel), CMD_UPDATE);
+}
+
+TSDLLEXPORT void
+ts_catalog_update_tid(Relation rel, ItemPointer tid, HeapTuple tuple)
+{
+	ts_catalog_update_tid_only(rel, tid, tuple);
 	/* Make changes visible */
 	CommandCounterIncrement();
```

Comment:
There seems to be a `CommandCounterIncrement` last in `ts_catalog_invalidate_cache`, so not sure why you need an extra one here.

Reply:
I don't think there's a `CommandCounterIncrement` in `ts_catalog_invalidate_cache`. But in any case, the `_only` functions should not increment the command counter. I fixed this issue for `_insert_only`